### ISSUE TITLE
Rule proposal: `prefer-object-from-entries`

### ISSUE THREAD

**Issue by fisker** [sindresorhus/eslint-plugin-unicorn#1260]
## Fail

```js
const foo = _.fromPairs(pairs);
```
Configurable function list like  https://github.com/sindresorhus/eslint-plugin-unicorn/blob/main/docs/rules/prefer-array-flat.md#functions

```js
const foo = pairs.reduce((object, [key, value]) => Object.assign(object, {[key]: value}), {});
```

```js
const foo = pairs.reduce((object, [key, value]) => Object.assign(object, {[key]: value}), Object.create(null));
```

```js
const foo = pairs.reduce((object, [key, value]) => ({...object, [key]: value}), {});
```

```js
const foo = pairs.reduce((object, [key, value]) => {
	object[key] = value;
	return object;
}, {});
```

```js
const foo = {};
for (const [key, value] of pairs) {
	foo[key] = value;
}
```

## Pass

```js
const foo = Object.fromEntries(pairs);
```

**Comment by bmish** [comment]
Oh, actually I just noticed that this ticket is for `fromEntries` but my idea is for `entries`. I'll file another ticket and delete this comment.

**Comment by sindresorhus** [comment]
This is now accepted.
